Add hostlogin to proxy config for windows desktop

rcanderson23 opened the pull request:
Resolves https://github.com/gravitational/cloud/issues/1635 and #12706 

When `HostLogin` is not set, it attempts to default to the username of the current linux user. This fails when running containers as an arbitrary UID and there is no matching user in `/etc/passwd`. This value doesn't appear to be used by Windows Desktop Services so it just needs to be set to something to avoid the issue.

rcanderson23 committed:
add hostlogin to proxy config for windows

zmb3 commented on lib/web/desktop.go:
```diff
@@ -193,6 +193,7 @@ func proxyClient(ctx context.Context, sessCtx *SessionContext, addr string) (*cl
 	if err := cfg.ParseProxyHost(addr); err != nil {
 		return nil, trace.Wrap(err)
 	}
+	cfg.HostLogin = sessCtx.user
```
Technically `HostLogin` should be the user on the target host, which we have in the `username` variable we pull from the query string on line 92.

It doesn't _really_ matter, as this is only used when using the `ProxyClient` to connect to an SSH node, but we do have more accurate information so we might as well use it.

I would also add a comment here:
```
// HostLogin must be specified in order to avoid defaulting to 
// the local Unix user, which will fail in some environments.
```

zmb3 replied:
Edit: I had time and realized it would be several hours before you're awake, so I went ahead and made the change. Let me know what you think!

zmb3 committed:
Set ProxyClient's HostLogin to the Windows username

Also convert a few TLS handshakes to a context-aware version
for better timeout/cancelation behavior.

zmb3 committed:
Merge branch 'master' into carson/windows-userid
